llarp/util/buffer.hpp

```diff
@@ -20,7 +20,84 @@
 namespace llarp
 {
   using byte_view_t = std::basic_string_view<byte_t>;
-}
+  using bstring = std::basic_string<std::byte>;
+  using bstring_view = std::basic_string_view<std::byte>;
+
+  struct llarp_buffer
+  {
+   private:
+    std::string _buf;
+    std::string_view _bview;
+    size_t _size;
+
+   public:
+    llarp_buffer() = default;
+    llarp_buffer(size_t s) : _size{s}
+    {
+      _buf.reserve(_size);
+      _bview = {_buf};
+    }
+    llarp_buffer(std::string b) : _buf{std::move(b)}, _bview{_buf}, _size{_buf.size()}
+    {}
+    llarp_buffer(std::string_view bv) : _buf{bv}, _bview{_buf}, _size{_buf.size()}
+    {}
+
+    template <
+        typename CharT,
+        std::enable_if_t<
+            std::is_convertible_v<CharT, char> || std::is_constructible_v<std::string, CharT>,
+            int> = 0>
+    llarp_buffer(CharT* c) : _buf{c}, _bview{_buf}, _size{_buf.size()}
+    {}
+
+    std::string_view
+    view() const
+    {
+      return _bview;
+    }
+
+    size_t
+    size() const
+    {
+      return _size;
+    }
+
+    bool
+    is_empty() const
+    {
+      return _buf.empty();
+    }
+
+    char*
+    data()
+    {
+      return _buf.data();
+    }
+    char*
+    data_at(size_t pos)
+    {
+      return _buf.data() + pos;
+    }
+
+    const char*
+    vdata()
+    {
+      return _bview.data();
+    }
+    const char*
+    vdata_at(size_t pos)
+    {
+      return _bview.data() + pos;
+    }
+
+    char
```

Comment:
should this be `char &` ? if it is not trying to get a reference it should be a `const` method maybe.